src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -1763,4 +1769,88 @@ namespace winrt::TerminalApp::implementation
+            {
+                return;
+            }
+            if (const auto& control{ p->GetTerminalControl() })
+            {
+                auto it = _controlEvents.find(*paneId);
+                if (it != _controlEvents.end())
+                {
+                    auto& events = it->second;
+
+                    if (isBroadcasting)
+                    {
+                        // ADD EVENT HANDLERS HERE
+                        events.keySentToken = control.KeySent([weakThis](auto&& sender, auto&& e) {
+                            if (const auto tab{ weakThis.get() })
+                            {
+                                if (tab->_tabStatus.IsInputBroadcastActive())
+                                {
+                                    if (const auto termControl{ sender.try_as<winrt::Microsoft::Terminal::Control::TermControl>() })
+                                    {
+                                        tab->_rootPane->BroadcastKey(termControl, e.VKey(), e.ScanCode(), e.Modifiers(), e.KeyDown());
+                                    }
+                                }
+                            }
+                        });
+
+                        events.charSentToken = control.CharSent([weakThis](auto&& sender, auto&& e) {
+                            if (const auto tab{ weakThis.get() })
+                            {
+                                if (tab->_tabStatus.IsInputBroadcastActive())
+                                {
+                                    if (const auto termControl{ sender.try_as<winrt::Microsoft::Terminal::Control::TermControl>() })
+                                    {
+                                        tab->_rootPane->BroadcastChar(termControl, e.Character(), e.ScanCode(), e.Modifiers());
+                                    }
+                                }
+                            }
+                        });
+
+                        events.stringSentToken = control.StringSent([weakThis](auto&& sender, auto&& e) {
+                            if (const auto tab{ weakThis.get() })
+                            {
+                                if (tab->_tabStatus.IsInputBroadcastActive())
+                                {
+                                    if (const auto termControl{ sender.try_as<winrt::Microsoft::Terminal::Control::TermControl>() })
+                                    {
+                                        tab->_rootPane->BroadcastString(termControl, e.Text());
+                                    }
+                                }
+                            }
+                        });
+                    }
+
+                    else
+                    {
+                        // REVOKE EVENT HANDLERS
+
+                        control.KeySent(events.keySentToken);
+                        control.CharSent(events.charSentToken);
+                        control.StringSent(events.stringSentToken);

```

Comment:
dumb question: do we properly revoke when you move pane to tab (!) or move tab to window (?)

Reply:
Sure seem to